sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/directconnectivity/rntbd/RntbdRequest.java

```diff
@@ -54,7 +55,7 @@ public static RntbdRequest decode(final ByteBuf in) {
             throw new IllegalStateException(reason);
         }
 
-        final int start = in.readerIndex();
+        final int start = in.writerIndex();
```

Comment:
why? is this related to this PR? or an unrelated bug?

Reply:
As mentioned in the original PR https://github.com/Azure/azure-sdk-for-java/pull/14697, we have to use writerIndex. 
Also moved this change to PR https://github.com/Azure/azure-sdk-for-java/pull/16197